src/oaklib/cli.py

```diff
@@ -202,13 +204,59 @@ def _shorthand_to_pred_curie(shorthand: str) -> PRED_CURIE:
         return shorthand
 
 
+# A list whose members are either strings (search terms, curies, or directives)
+# or nested lists.
+# TODO: Replace this with an explicit query model with boolean operations
+NESTED_LIST = Union[List[str], List["NESTED_LIST"]]
+
+
+def nest_list_of_terms(terms: List[str]) -> NESTED_LIST:
+    """
+    Gives a list of terms (typically passed on command line),
+    replace blocks between '[', ..., ']' with nested lists of the contents
+
+    :param terms:
+    :return:
+    """
+    nested, rest = _nest_list_of_terms(terms)
+    if rest:
+        raise ValueError(f"Unparsed: {rest}")
+    return nested
+
+
+def _nest_list_of_terms(terms: List[str]) -> Tuple[NESTED_LIST, List[str]]:
+    nested = []
+    while len(terms) > 0:
+        term = terms[0]
+        terms = terms[1:]
```

Comment:
minor but can't lines 230 and 231 be replaced by `term = terms.pop(0)`

Reply:
```
term = terms[0]
terms = terms[1:]
```

can be replaced with clever usage of the splat operator

```
term, *terms = terms
```